homeassistant/components/switch/hlk_sw16.py

```diff
@@ -0,0 +1,43 @@
+"""
+Support for HLK-SW16 switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.hlk-sw16/
+"""
+import logging
+
+from homeassistant.components.hlk_sw16 import (
+    SwitchableSW16Device, DOMAIN as HLK_SW16)
+from homeassistant.components.switch import (
+    ToggleEntity)
+from homeassistant.const import CONF_NAME
+
+DEPENDENCIES = [HLK_SW16]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def devices_from_config(domain_config):
+    """Parse configuration and add HLK-SW16 switch devices."""
+    devices = []
+    for device_port, device_name in domain_config[HLK_SW16]:
+        device = SW16Switch(device_name.get(CONF_NAME, device_port),
+                            device_port)
+        devices.append(device)
+    return devices
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the HLK-SW16 platform."""
+    async_add_entities(devices_from_config(discovery_info),
+                       update_before_add=True)
+
+
+class SW16Switch(SwitchableSW16Device, ToggleEntity):
+    """Representation of a HLK-SW16 switch."""
+
+    @property
+    def force_update(self) -> bool:
+        """Will trigger anytime the state property is updated."""
+        return True
```

Comment:
Why is it useful to force update for a switch entity? If the entity has the same state as before when an update comes in, we shouldn't create a new state. Forced updates is used for sensors where reporting the same state is useful, eg a temperature tracking sensor.

Reply:
I think it's not needed in that case, I'll remove it.

Reply:
removed [here](https://github.com/home-assistant/home-assistant/pull/17855/commits/743c244c23ef0ae33a935be73277bb36a327f25e)